### DISCUSSION TITLE
Second user same library

### DISCUSSION THREAD

**Discussion by DeltaTango69** [immich-app/immich#11010] [Q&A]
So far I have used Immich on my own. Now I would like to create an account for my wife as well. However, it is a little unclear to me how a new user is handled. I would like to access the same external libraries with both users. Of course, thumbs should only be created once for this. As far as I understand it, they would be created again for a second user, right? The photos from the smartphone should be saved in different directories. But all users should also see all pictures from all users and the shared external libraries too, of course. How can I manage this?

**Comment by bo0tzz** [comment]
Sharing a library is not possible at the moment.

**Reply by DeltaTango69** [reply]
Is this planned for the future, maybe?

What happens if both users log in to the smartphone with the same account? Are the images mixed and written to the same upload directory?

**Reply by bo0tzz** [reply]
We're currently reworking the permissions/access control systems, hopefully this will be made possible by that yes.

Indeed, if you share an account and enable upload then the assets are mixed.

**Reply by eigan** [reply]
@bo0tzz Is there any pr/issue tracking these changes you are talking about here?

**Comment by tarkh** [comment]
Hello! Thank you for an excellent product!

It would be great to have a permission system to better organize family access. This could be implemented using standard read/write permissions for users and groups. For example, I could create a `family` group, assign read/write permissions and add my wife to it. Or directly assign permissions to a user. This is a straightforward way to manage access.

Sharing, on the other hand, could remain focused on publicly sharing specific albums with friends.

**Reply by axgd-code** [reply]
there are two subjects:
- Fine grained access controls (Granular access controls for users and api keys) which is planned in 2025 (cf. roadmap)
- https://github.com/immich-app/immich/discussions/13621 multi-instances of immich to share with others people
